### ISSUE TITLE
Trying to assign stateChanged to a type Subscription is causing an error

### ISSUE THREAD

**Issue by jonpfl** [DanWahlin/Observable-Store#24]
Dan,

When I do the following code

`import { Observable, forkJoin, SubscriptionLike, Subscription } from 'rxjs';`
`storeSub: Subscription;`

`  ngOnInit() {`
`    this.wait('Please wait while the group data is loaded.');`
`    this.loadGroups();`
`    this.storeSub = this.groupMeetingsService.stateChanged.subscribe(state => {`
`      console.log('state changed', state);`
`    });`
`  }`

this.storeSub is saying 

(property) GroupFunctionsComponent.storeSub: Subscription
Type 'Subscription' is missing the following properties from type 'Subscription': _parent, _parents, _addParentts(2739)

I can use SubscriptionLike and the problem goes away.

Thx
jonpfl

**Comment by elAndyG** [comment]
could you share a snippet of your groupMeetingService file or maybe a stackblitz with a sample of what you are trying to do? I'm curious if there is a setup issue. I tried to duplicate your code and was able to set my `storeSub` and destroy it ok.

**Comment by jonpfl** [comment]
Andy,

Sorry, I would like to share but not sure how my company would feel about me putting proprietary code on the web, I hope you understand.

If/when I get back to this, if I am still having an issue, I will revisit the proof of concept code and let you know.

Thx
jonpfl

**Comment by elAndyG** [comment]
no worries @jonpfl , i created a simple stackblitz with a subscription. Hopefully it helps out. 

https://stackblitz.com/edit/simple-observable-store-example?file=src%2Fapp%2Fhello.component.ts

**Comment by DanWahlin** [comment]
Thanks for putting that together @elAndyG! @jonpfl and I suspect it may be an RxJS version issue. Although I'm not able to duplicate the issue at this point, I'm going to leave it open in case anyone else runs into it so that I can get some additional information on what might be causing it. Thanks again for sharing the stackblitz demo.

**Comment by DanWahlin** [comment]
@jonpfl I'm going to go ahead and close this one since we'll need a sample or project to run in order to duplicate the problem you were seeing. I tried multiple things to try to simulate it but never got the specific issue. I still think it's likely related to different versions of RxJS being in a project - but not positive on that.

If you run into it again and have a sample (even if very simple) let us know and we can create a new issue for it if needed. Thanks!

**Comment by jonpfl** [comment]
Dan,

No problem, when/if I get back to messing around with it, I will let you
know.

Thx for all your help!!
jonpfl

**Comment by DanWahlin** [comment]
Sounds great - thanks!

**Comment by bartvanraaij** [comment]
Hi, I encountered the same problem:
```
[ng] ERROR in src/app/models/request-store.ts(16,5): error TS2322: Type 'import("/Users/bart/Dev/smeba-app/node_modules/@codewithdan/observable-store/node_modules/rxjs/internal/Subscription").Subscription' is not assignable to type 'import("/Users/bart/Dev/smeba-app/node_modules/rxjs/internal/Subscription").Subscription'.
[ng]   Property '_parent' is missing in type 'Subscription'.
```

I fixed it by changing rxjs version `6.3.3` to `^6.5.2` in `package.json` and running `npm update rxjs`

**Comment by DanWahlin** [comment]
Thanks @bartvanraaij. Really appreciate the information there. 

That confirms my suspicion about an RxJS version conflict. I'm wondering if moving the Observable Store RxJS version down to a lower version to accommodate older versions of RxJS in things like Angular would help.  I'll have to look into that as soon as I get a chance.
